dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/compute/function/DatacenterToLocationTest.java

```diff
@@ -0,0 +1,95 @@
+import org.jclouds.dimensiondata.cloudcontrol.domain.Monitoring;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Networking;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Property;
+import org.jclouds.domain.Location;
+import org.jclouds.domain.LocationBuilder;
+import org.jclouds.domain.LocationScope;
+import org.jclouds.location.suppliers.all.JustProvider;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+import static org.easymock.EasyMock.createNiceMock;
+import static org.easymock.EasyMock.expect;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "unit", testName = "DatacenterToLocationTest")
+public class DatacenterToLocationTest {
+
+   private Datacenter datacenter;
+   private JustProvider justProvider;
+   private Set parentLocations;
+
+   @BeforeTest
+   public void setUp() {
+      final List<Hypervisor.DiskSpeed> diskSpeeds = Collections.singletonList(
+            Hypervisor.DiskSpeed.create("STANDARD", "Standard", "STD", "Standard Disk Speed", true, true));
+      final List<Hypervisor.CpuSpeed> cpuSpeeds = Collections
+
+            .singletonList(Hypervisor.CpuSpeed.create("STANDARD", "Standard", "Standard CPU Speed", true, true));
+      final Hypervisor hypervisor = Hypervisor.builder().diskSpeeds(diskSpeeds).cpuSpeeds(cpuSpeeds)
+            .properties(Collections.singletonList(Property.create("MIN_DISK_SIZE_GB", "10")))
+            .maintenanceStatus("NORMAL").type("VMWARE").build();
+
+      parentLocations = ImmutableSet.of(new LocationBuilder().scope(LocationScope.PROVIDER).id("parentId")
+            .iso3166Codes(new ArrayList<String>()).metadata(new HashMap()).description("parentDescription").build());
+
+      datacenter = Datacenter.builder().displayName("US - West").city("Santa Clara").state("California").country("US")
+            .vpnUrl("https://na3.cloud-vpn.net").ftpsHost("ftps-na.cloud-vpn.net").networking(Networking.builder()
+                  .properties(Collections.singletonList(Property.create("MAX_SERVER_TO_VIP_CONNECTIONS", "20")))
+                  .type("1").maintenanceStatus("NORMAL").build()).hypervisor(hypervisor).backup(
+                  Backup.builder().maintenanceStatus("NORMAL").type("COMMVAULT")
+                        .properties(Collections.<Property>emptyList()).build()).consoleAccess(
+                  ConsoleAccess.builder().properties(Collections.<Property>emptyList()).maintenanceStatus("NORMAL")
+                        .build()).monitoring(
+                  Monitoring.builder().maintenanceStatus("NORMAL").properties(Collections.<Property>emptyList())
+                        .build()).type("MCP 1.0").id("NA3").build();
+
+      justProvider = createNiceMock(JustProvider.class);
```

Comment:
do you need to mock the `justProvider` or you can maybe re-use the pattern used at https://github.com/jclouds/jclouds/blob/master/providers/digitalocean2/src/test/java/org/jclouds/digitalocean2/compute/functions/RegionToLocationTest.java#L41-L42 ?